### PR TITLE
[Merged by Bors] - chore: cleanup using lake exe cache clean

### PR DESCRIPTION
**Pull request by kim-em**
This moves CI back to the Hoskinson machines (after being disabled temporarily in https://github.com/leanprover-community/mathlib4/pull/4726), but cleans up the cache before each build.

~~(We can't merge this until we manually clean up!)~~

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .github/workflows/bors.yml [hunk 1]
```diff
@@ -108,7 +108,9 @@ jobs:
           find Mathlib -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
 
       - name: get cache
-        run: lake exe cache get
+        run: |
+          lake exe cache clean!
```
Comment (semorrison) [inline .github/workflows/bors.yml:112]:
```suggestion
          lake exe cache clean
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -114,7 +114,9 @@ jobs:
           find Mathlib -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
 
       - name: get cache
-        run: lake exe cache get
+        run: |
+          lake exe cache clean!
```
Comment (semorrison) [inline .github/workflows/build.yml:118]:
```suggestion
          lake exe cache clean
```

File: .github/workflows/build.yml.in [hunk 3]
```diff
@@ -94,7 +94,9 @@ jobs:
           find Mathlib -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
 
       - name: get cache
-        run: lake exe cache get
+        run: |
+          lake exe cache clean!
```
Comment (semorrison) [inline .github/workflows/build.yml.in:98]:
```suggestion
          lake exe cache clean
```

File: .github/workflows/build_fork.yml [hunk 4]
```diff
@@ -112,7 +112,9 @@ jobs:
           find Mathlib -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
 
       - name: get cache
-        run: lake exe cache get
+        run: |
+          lake exe cache clean!
```
Comment (semorrison) [inline .github/workflows/build_fork.yml:116]:
```suggestion
          lake exe cache clean
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**